lib/stat_tracker.rb

```diff
@@ -1,13 +1,90 @@
 class StatTracker
-  attr_reader :locations
+  attr_reader :files,
+              :game_stats,
+              :team_stats,
+              :league_stats
 
-  def initialize(locations)
-    @locations = locations
-    #@game_info = file_converter.new(locations[:games]).game_info_converter
+  def initialize(files)
+    @files = files
+    @game_stats = compile_game_stats
+    @team_stats = compile_team_stats
+    @league_stats = compile_league_stats
   end
 
-  def self.from_csv(locations)
-    new(locations)
+  def self.from_csv(files)
+    new(files)
+  end
+
+  # StatCompiler superclass
+  # ======================================
+  def convert_files
+    FileConverter.new(files)
+  end
+
+  def compile_game_stats
+    games = convert_files.games
+    GameStats.new(games)
+  end
+
+  def compile_team_stats
+    teams = convert_files.teams
+    TeamStats.new(teams)
+  end
```

Comment:
These could be combined into one method and called at the same time, but makes no difference so fine to keep separated.

Reply:
I agree, I kept them separated to access the method in instance variables, but I could flip that and put the instance variables @game_stats and @team_stats in a method compile_stats and that method as an instance variable in initialize. I'm going to push a new version up that is updated from the last merge with season_stats and can make that change.